router/transformer/transformer.go

```diff
@@ -186,19 +186,31 @@ func (trans *HandleT) ResponseTransform(ctx context.Context, responseData integr
 		payload.Seek(0, io.SeekStart)
 		req, err := http.NewRequestWithContext(ctx, "POST", url, payload)
 		if err != nil {
-			trans.logger.Error(fmt.Sprintf(`400 Unable to construct POST request for URL : "%s"`, url))
-			return 400, fmt.Sprintf(`400 Unable to construct POST request for URL : "%s"`, url)
+			errStr := fmt.Sprintf(`400 Unable to construct POST request for URL : "%s"`, url)
+			trans.logger.Error(errStr)
+			return http.StatusBadRequest, errStr
 		}
 		req.Header.Add("Content-Type", "application/json")
 		s := time.Now()
 		resp, err = trans.client.Do(req)
 		trans.transformerResponseTransformRequestTime.SendTiming(time.Since(s))
-		// retry in case of err
-		if err != nil {
-			trans.logger.Errorf("[Transformer Response Transform request failed] :: %+v", err)
+
+		//Handle for error cases for request to transformer, we expect the transformer to retrun 200
+		//for successful response
+		if err != nil || (resp != nil && resp.StatusCode != http.StatusOK) {
+			var tStatus string
+			if resp != nil {
+				tStatus = resp.Status
+				if resp.StatusCode == http.StatusNotFound {
+					panic(fmt.Errorf("[Response transform doesnot exist for URL: URL: %v", url))
+				}
+
+			}
```

Comment:
This part or code is a bit hard to read/follow. Also, it is not very idiomatic, usually, we check for err != nil before doing anything else, but checks here are a bit convoluted.

May I suggest a more idiomatic way in combination with if-else or early returnn

```go

if err != nil {
    // ...
	retryCount++
	time.Sleep(retrySleep)
    continue
}

// if err == nil, resp is always set

if resp.StatusCode == http.StatusNotFound {
      panic(fmt.Errorf("[Response transform doesnot exist for URL: URL: %v", url))
}

if resp.StatusCode != http.StatusOK {
	// check of other conditions here	   
	retryCount++
	time.Sleep(retrySleep)
	continue
}

if resp.Body == nil {
  // error handling if body is missing 
}

tempRespData, err = io.ReadAll(resp.Body)
if err != nil {

}




```

Reply:
Yeah makes sense, will address this

Reply:
The motivation behind writing the code the way it's written

We retry when there is a problem with transformer service. Hence we've kept the transformer service error handling at a single place.
Also this would avoid code duplication
```golang

 if err != nil || (resp.StatusCode != 200) {
   // There is prob with transformer service
 } 
```
Idiomatic perspective I understand. Want to understand why do you think it's not readable. Can you shed more light on this ?